tests/InstrEngineTests/NaglerInstrumentationMethod/NaglerInstrumentationMethod.cpp

```diff
@@ -20,15 +20,24 @@ HRESULT CInstrumentationMethod::Initialize(_In_ IProfilerManager* pProfilerManag
 
     DWORD retVal = WaitForSingleObject(hConfigThread, 15000);
 
-    if (m_bExceptionTrackingEnabled)
+    if (m_bTestInstrumentationMethodLogging)
+    {
+        pProfilerManager->GetLoggingInstance(&m_pProfilerManagerInstanceLogger);
+
+        CComQIPtr<IProfilerManager4> pProfilerManager4 = pProfilerManager;
+        pProfilerManager4->GetGlobalLoggingInstance(&m_pProfilerManagerLogger);
+        m_pProfilerManagerLogger->LogDumpMessage(_T("<InstrumentationMethodLog>"));
+        m_pProfilerManagerInstanceLogger->LogDumpMessage(_T("Success!"));
+    }
+    else if (m_bExceptionTrackingEnabled)
     {
         CComPtr<ICorProfilerInfo> pCorProfilerInfo;
         pProfilerManager->GetCorProfilerInfo((IUnknown**)&pCorProfilerInfo);
         pCorProfilerInfo->SetEventMask(COR_PRF_MONITOR_EXCEPTIONS | COR_PRF_ENABLE_STACK_SNAPSHOT);
 
-        CComPtr<IProfilerManagerLogging> spLogger;
-        pProfilerManager->GetLoggingInstance(&spLogger);
-        spLogger->LogDumpMessage(_T("<ExceptionTrace>\r\n"));
+        CComQIPtr<IProfilerManager4> pProfilerManager4 = pProfilerManager;
+        pProfilerManager4->GetGlobalLoggingInstance(&m_pProfilerManagerLogger);
```

Comment:
nit: If you end up keeping this change, please rename to something like m_pGlobalLogger

Reply:
Reverting my change back to the original code with spLogger instead.

---
In reply to: [341220900](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r341220900) [](ancestors = 341220900)